r/R/metadata.R

```diff
@@ -142,7 +142,8 @@ arrow_attributes <- function(x, only_top_level = FALSE) {
   }
 
   columns <- NULL
-  if (is.list(x) && !inherits(x, "POSIXlt")) {
+  attempt_to_save_row_level <- getOption("arrow.preserve_row_level_metadata", FALSE)
```

Comment:
I'm a bit hesitant to totally rip out the ability to do this (especially right before a release) in case there are folks who depend on this feature. 

When I looked at https://issues.apache.org/jira/browse/ARROW-10386 it's clear that some people do want this. For SF, we can recommend using {sfarrow} of course, but for anything else without this option there would be no way to use Arrow with that data without losing this metadata.

This also makes it slightly easier to test that we can still read in the metadata without needing to write a golden file that has the metadata and curating that, since we can enable this for writting and then read in data and see that it's warning/dropping where appropriate.